scripts/zkai_attention_kv_softmax_denominator_rounding_edge_corpus_gate.py

```
@@ -0,0 +1,478 @@
+        "max_denominator": max(denominators),
+        "max_remainder_ratio": max_remainder_ratio,
+        "negative_numerator_cases": [
+            case["name"] for case in cases if case["negative_numerator_dimensions"] > 0
+        ],
+        "validation_commands": list(VALIDATION_COMMANDS),
+    }
+    validate_result(result)
+    return result
+
+
+def validate_result(result: Any) -> None:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `result`

(ANN401)